### PR TITLE
Add Github Actions CI

### PR DESCRIPTION
**Pull request by rachelekm**
## Overview

This PR adds support for Github Actions and runs `cibuild` as part of CI workflow. Workflow is triggered on push event and does not include support for deployment/publishing images to ECR.

Connects #14 

## Testing Instructions

 * Push a dummy commit to this branch and confirm that it triggers a build
 * Confirm that the CI workflow passes

### CODE REVIEW

File: scripts/cibuild [hunk 1]
```diff
@@ -23,6 +23,7 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
     if [ "${1:-}" = "--help" ]; then
         usage
     else
+        ./scripts/format
         ./scripts/lint
```
Comment (rachelekm) [inline scripts/cibuild:26]: Instead of scripts/format, scripts/lint includes check for black formatting. However, scripts/lint doesn't include prettier format checking in javascript.

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -5,4 +5,4 @@ jobs:
     name: build
     runs-on: ubuntu-latest
     steps:
-      - run: .scripts/cibuild
+      - run: ./scripts/cibuild
```
Comment (rachelekm) [inline .github/workflows/continuous_integration.yml:9]: fixup this commit with previous - every commit should be in functional state of the project.

File: scripts/cibuild [hunk 3]
```diff
@@ -23,16 +23,33 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
...
-            run --rm --no-deps \
-            -e NODE_ENV="${NODE_ENV:-production}" \
-            -e INSTALL_ENV="${INSTALL_ENV:-internal}" \
-            -e REACT_APP_GIT_COMMIT="${GIT_COMMIT}" client \
-            yarn build
+            -f docker-compose.yml \
+            up -d database
+
+        # Build tagged container images
+        GIT_COMMIT="${GIT_COMMIT}" docker-compose \
```
Comment (rachelekm) [inline scripts/cibuild]: Do not need to set GIT_COMMIT var since it's not being used in the following script files

File: scripts/cibuild [hunk 4]
```diff
@@ -23,16 +23,33 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
...
-        ./scripts/test
+        #Update dependencies
+        echo "Updating frontend dependencies"
+        docker-compose \
+            run --rm client \
+            yarn install
+
+        #Build static asset bundle
+        echo "Building static bundle"
+        docker-compose \
```
Comment (rachelekm) [inline scripts/cibuild]: Unsure of this purpose: worth taking out and seeing if it breaks?

File: django/api/models.py [hunk 5]
```diff
@@ -20,7 +20,9 @@ def average_rating(self):
 
 class Rating(models.Model):
     Stars = models.IntegerChoices("star", "one two three four five")
-    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    user = models.ForeignKey(
```
Comment (rachelekm) [inline django/api/models.py]: Configure editor so that no longer need noqa comments, and squash so that this commit is eventually negated (after new commit with flake8 changes -  look into universal editor config file)

File: client/package.json [hunk 6]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "dependencies": {
     "@fortawesome/fontawesome-svg-core": "^1.2.36",
+    "@types/jest": "^27.0.2",
```
Comment (rachelekm) [inline client/package.json]: worth noting: this commit includes formatting/linting logic that is better squashed into previous commit's linting/formatting logic. Better to separate dependency update into its own commit and bring together all the format/lint logic into one separate commit (so that when I do make flake8 change, everything that can react from one commit)

File: client/src/setupTests.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        this.addControl = mockMapaddControl;
+        this.touchZoomRotate = mocktouchZoomRotate;
+        this.doubleClickZoom = mockdoubleClickZoom;
+
+        this.dragRotate.disable = jest.fn();
+        this.touchZoomRotate.disableRotation = jest.fn();
+        this.doubleClickZoom.disable = jest.fn();
+    },
+    LngLat: mockLngLat,
+    LngLatBounds: mockLngLatBounds,
```
Comment (ddohler) [inline client/src/setupTests.js:38]: I'm noticing that although the tests are passing, there's a [console error in the logs](https://github.com/rachelekm/vegishphilly/pull/15/checks?check_run_id=3898417186#step:3:186) . I'm not clear why it's not causing the test to fail, but it should be fixed one way or another (looks like it just needs another mock).

**Review by ddohler — COMMENTED** [review]
This looks really good, nice job! I did notice a console error in the logs that strangely isn't causing the tests to crash, so that should get resolved before merging, but otherwise this looks great.

File: client/src/setupTests.js (hunk 7 above)
Comment (rachelekm) [reply client/src/setupTests.js:38]: Ah I missed these, but should be easy enough to mock. Thank you!

**Review by ddohler — APPROVED** [review]
Looking good, great job! 👏
